interp/interp.go

```diff
@@ -0,0 +1,165 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package interp
+
+import (
+	"fmt"
+	"math"
+	"sort"
+
+	"gonum.org/v1/gonum/spatial/r1"
+)
+
+// Interpolator1D represents a 1D interpolator.
+// It interpolates a y(x) function based on some data over a range [begin(), end()].
+// Both begin() and end() can be infinite, but begin() must be lower than end().
+type Interpolator1D interface {
+
+	// Returns the closed interval over which one can interpolate.
+	Interval() r1.Interval
+
+	// Evaluates interpolated sequence at x in [begin(), end()].
```

Comment:
```
// Eval returns the interpolation value at x.
// It panics if x is not in the interpolation interval.
Eval(x float64) float64
```

Bikeshed: `ValueAt`, `Value`? instead of `Eval`.

Reply:
I'll go for `ValueAt`.